Sekoia.io/sekoiaio/operation_center/synchronize_assets_with_ad.py

```diff
@@ -1,24 +1,33 @@
 from urllib.parse import urljoin
-from typing import List, Dict, Any
+from typing import List, Dict, Any, Optional
 import requests
 import json
 from pydantic import BaseModel
 from sekoia_automation.action import Action
 
 
 class Arguments(BaseModel):
-    user_ad_data: Dict[str, Any]
+    user_ad_data: Optional[Dict[str, Any]] = None
     asset_synchronization_configuration: Dict[str, Any]
     community_uuid: str
+    user_ad_file: Optional[str] = None
 
 
 class SynchronizeAssetsWithAD(Action):
     """
     Action to synchronize asset with Active Directory (AD).
     """
 
-    def run(self, arguments: Arguments):
-        user_ad_data = arguments.user_ad_data
+    def run(self, arguments: Arguments) -> List[Dict[str, Any]]:
+        # Determine if user_ad_data is a list
+        if arguments.user_ad_file is not None:
+            user_ad_data = self.json_argument("user_ad_file", arguments)
+        else:
+            user_ad_data = arguments.user_ad_data
```

Comment:
`json_argument` works with dictionary, not object. So, it won't work here.
Sadly, the simplest thing to do here is to convert back `arguments` to a dict :(

```suggestion
    def run(self, arguments: dict) -> List[Dict[str, Any]]:
        user_ad_data = self.json_argument("user_ad_data", arguments)
```

Reply:
Done